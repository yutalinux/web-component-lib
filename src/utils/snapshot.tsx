import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
