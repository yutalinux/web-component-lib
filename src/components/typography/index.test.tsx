import { snapshotElement } from "@/utils/snapshot";
import { expect, test } from "vitest";
import { Typography } from ".";

test("Typography Snapshot", () => {
  expect(
    snapshotElement(Typography, { type: "h1", label: "Heading1" }),
  ).toMatchSnapshot();
  expect(
    snapshotElement(Typography, { type: "h2", label: "Heading2" }),
  ).toMatchSnapshot();
  expect(
    snapshotElement(Typography, { type: "h3", label: "Heading3" }),
  ).toMatchSnapshot();
  expect(
    snapshotElement(Typography, { type: "h4", label: "Heading4" }),
  ).toMatchSnapshot();
  expect(
    snapshotElement(Typography, { type: "h5", label: "Heading5" }),
  ).toMatchSnapshot();
  expect(
    snapshotElement(Typography, { type: "h6", label: "Heading6" }),
  ).toMatchSnapshot();
  expect(
    snapshotElement(Typography, { type: "p", label: "Heading6" }),
  ).toMatchSnapshot();
});
