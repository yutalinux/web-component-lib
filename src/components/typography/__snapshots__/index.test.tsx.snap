// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToggleButton Snapshot 1`] = `"<h1>Heading1</h1>"`;

exports[`ToggleButton Snapshot 2`] = `"<h2>Heading2</h2>"`;

exports[`ToggleButton Snapshot 3`] = `"<h3>Heading3</h3>"`;

exports[`ToggleButton Snapshot 4`] = `"<h4>Heading4</h4>"`;

exports[`ToggleButton Snapshot 5`] = `"<h5>Heading5</h5>"`;

exports[`ToggleButton Snapshot 6`] = `"<h6>Heading6</h6>"`;
